code/game/objects/items/devices/traitordevices.dm

```diff
@@ -276,14 +276,16 @@ effective or pretty fucking useless.
 	new /obj/item/wirecutters(src)
 
 /obj/item/storage/toolbox/emergency/turret/attackby(obj/item/attacking_item, mob/living/user, params)
-	if(attacking_item.tool_behaviour == TOOL_WRENCH && user.combat_mode && attacking_item.use_tool(src, user, 2 SECONDS, volume = 50))
-		user.visible_message(span_danger("[user] bashes [src] with [attacking_item]!"), \
-			span_danger("You bash [src] with [attacking_item]!"), null, COMBAT_MESSAGE_RANGE)
-		playsound(src, "sound/items/drill_use.ogg", 80, TRUE, -1)
-		var/obj/machinery/porta_turret/syndicate/toolbox/turret = new(get_turf(loc))
-		set_faction(turret, user)
-		turret.toolbox = src
-		forceMove(turret)
+	if(istype(attacking_item, /obj/item/wrench/combat))
+		if(user.combat_mode && attacking_item.toolspeed && attacking_item.use_tool(src, user, 5 SECONDS, volume = 20))
```

Comment:
why not use early returns here? much easier and more readable

Reply:
I based it on deconstruction code
/obj/machinery/porta_turret/syndicate/toolbox/attackby(obj/item/attacking_item, mob/living/user, params)
	if(istype(attacking_item, /obj/item/wrench/combat))
		if(user.combat_mode && attacking_item.toolspeed && attacking_item.use_tool(src, user, 5 SECONDS, volume = 20))
			deconstruct(TRUE)
			attacking_item.play_tool_sound(src, 50)
		else if(!user.combat_mode)
			to_chat(user, span_notice("You start repairing [src]..."))
			while(atom_integrity != max_integrity && attacking_item.toolspeed && attacking_item.use_tool(src, user, 2 SECONDS, volume = 20))
				repair_damage(10)
		return
	return ..()

Reply:
![5294536ab0aa0320616cf364ad83bfa2](https://user-images.githubusercontent.com/47064429/180099546-04bdf195-9335-4155-b44f-dc798b15abc2.png)

Reply:
That code either has a better reason for not using early returns, or is old. If you use early returns you can make it a lot simpler to read and easier to debug. ie:

```
proc/hunger()
    if(is_hungry)
         if(has_food)
            if(likes_food)
                 eat_food()
    return
```
doesn't parse visually very well compared to
```
proc/hunger()
   if(!is_hungry)
      return

   if(!has_food)
       return

   if(!likes_food)
       return

   eat_food()
```

Reply:
It was done 3 months ago by Fikou so im really not sure

Reply:
Also how it would look with else if too?

Reply:
`else` only passes if the preceding `if` doesn't. `else` is used for when you have multiple checks in succession that aren't mutally exclusive, but you want to stop running if a certain one passes.

Since the first `if` checks `user.combat_mode`, if it's false, it will always fail. So, the `else if` will always pass. They're not mutally exclusive, so there's no reason to have `else` here, though it works. So feel free to get rid of the `else` and just have `if(!user.combat_mode)`.

Reply:
Ok. I will change it in the evening after I came back from work